sys/kern/src/kipc.rs

```diff
@@ -378,22 +432,40 @@ fn read_task_dump_region(
             // memory, or wholly bogus).  In all of these cases, we will
             // rely on `safe_copy` to do the validation.
             //
-            let from = match USlice::<u8>::from_raw(
-                region.base as usize,
-                region.size as usize,
-            ) {
-                Ok(from) => from,
-                Err(err) => {
-                    return Err(UserError::Unrecoverable(
-                        FaultInfo::SyscallUsage(err),
-                    ));
-                }
-            };
+            // Note that this applies to: attempts to read out of bounds of the
+            // target task's TCB, attempts to read other tasks' TCBs, etc. So
+            // this is effectively serving as the validation backstop for the
+            // TCB access code above.
+            //
 
-            match safe_copy(tasks, index as usize, from, caller, response) {
+            match safe_copy(tasks, index, from, caller, response) {
                 Err(interact) => {
-                    let wake_hint = interact.apply_to_dst(tasks, caller)?;
-                    return Ok(wake_hint);
+                    // So, this is a weird case. We are attempting to transfer
+                    // memory from one task to another, but unlike every other
+                    // use of safe_copy in the kernel, the address ranges on
+                    // _both sides_ are controlled by one party (the
+                    // supervisor).
+                    //
+                    // This means the InteractFault produced by `safe_copy` is
+                    // shaped slightly wrong for the task. Counter-intuitively,
+                    // the supervisor (meaning `caller`) should be blamed for
+                    // faults on _both_ the source and destination side.
+                    //
+                    // So we do this using the following code, which looks wrong
+                    // at first glance since it's backwards from every other
+                    // routine in the kernel -- but its shape is deliberate! We
+                    // really do want to apply the SRC fault to the caller,
+                    // followed by any remaining DST fault.
```

Comment:
Maybe add a comment here explaining that the `DST` fault is applied all the way at the top of the call stack in `safe_syscall_entry`, since I had to search for that.

Reply:
done!